types/mempool.go

```diff
@@ -0,0 +1,183 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/google/btree"
```

Comment:
https://github.com/tidwall/btree may be an interesting alternative?

Reply:
maybe.. a quick scan reveals I can't alter the degree (hard-coded to 128).  I opted for a 2-3-4 (red-black) tree here, but we could run some benchmarks at different degrees.